bin/node-template/pallets/template/src/mock.rs

```diff
@@ -48,6 +48,7 @@ impl system::Trait for Test {
 	type AccountData = ();
 	type OnNewAccount = ();
 	type OnKilledAccount = ();
+	type SystemWeightInfo = ();
```

Comment:
AFAICT this only resolves the conflict for the common case of `Trait: system::Trait`, but not for `Trait: balances::Trait`.

Reply:
Who uses `balances::Trait` directly? They should probably be using `Currency` right?

In either case the solution in general just becomes `T::WeightInfo` -> `<T as crate_name::Trait>::WeightInfo`, so totally solvable in general, but ideally the first syntax works on average :)

Reply:
I meant `balances::Trait` more as a stand-in for people using direct dependency on other pallets which we (want to?) support.

Reply:
Naming everything uniquely would work here, but I also think there is a lot of clarity and simplicity to gain from this uniformity. In general, hard dependencies on other pallet traits is not recommended. But if a user opts to do this, then the solution above is correct, and I think a reasonable place to be

Reply:
ok, agreed :-)